modules/nextflow/src/main/groovy/nextflow/cloud/aws/batch/AwsBatchFileCopyStrategy.groovy

```diff
@@ -93,7 +93,12 @@ class AwsBatchFileCopyStrategy extends SimpleFileCopyStrategy {
     @Override
     String stageInputFile( Path path, String targetName ) {
         // third param should not be escaped, because it's used in the grep match rule
-        "downloads+=(\"nxf_s3_download s3:/${Escape.path(path)} ${Escape.path(targetName)}\")"
+        if (opts.maxTransferAttempts > 1) {
+            "downloads+=(\"nxf_s3_retry nxf_s3_download s3:/${Escape.path(path)} ${Escape.path(targetName)}\")"
+        }
+        else {
+            "downloads+=(\"nxf_s3_download s3:/${Escape.path(path)} ${Escape.path(targetName)}\")"
+        }
     }
```

Comment:
Instead of using an `if` branch, the code would result more readable writing something like: 

```
def cmd = opts.maxTransferAttempts > 1 ? "nxf_s3_retry nxf_s3_download" : "nxf_s3_download"
"downloads+=(\"$cmd s3:/${Escape.path(path)} ${Escape.path(targetName)}\")"
```

Reply:
Sure. I changed also. Also I thought it might be worth adding unit test for this, but now some test from HttpFilesTests failed after I added the new one. Locally they are still passing. Do you have any idea why this is happening now?

Reply:
I think it was a temporary network hiccup, relaunching it the build tests are green